lib/cmd_range2ip.go

```diff
@@ -0,0 +1,98 @@
+package lib
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+// CmdRange2IPFlags are flags expected by CmdRange2IP.
+type CmdRange2IPFlags struct {
+	Help bool
+}
+
+// Init initializes the common flags available to CmdRange2IP with sensible
+// defaults.
+//
+// pflag.Parse() must be called to actually use the final flag values.
+func (f *CmdRange2IPFlags) Init() {
+	pflag.BoolVarP(
+		&f.Help,
+		"help", "h", false,
+		"show help.",
+	)
+}
+func CmdRange2IP(f CmdRange2IPFlags, args []string, printHelp func()) error {
+	if f.Help {
+		printHelp()
+		return nil
+	}
+
+	// require args and/or stdin.
+	stat, _ := os.Stdin.Stat()
+	isStdin := (stat.Mode() & os.ModeCharDevice) == 0
+	if len(args) == 0 && !isStdin {
+		printHelp()
+		return nil
+	}
+	// if gets piped input
```

Comment:
All of the code below is a hack of just copy-pasting code from `lib.IPListWriteFrom`.  We should be modifying `lib.IPListWriteFrom` itself to allow us to express what you're trying to do here.

Specifically, `lib.IPListWriteFrom` currently uses `ip`, `iprange` and `cidr` as booleans to modify how we parse `inputs`. So if we want to only parse `iprange`s from `inputs`, we simply set `iprange=true` and leave the others as `false`. Everything works the way we want in that case.

But for parsing files and stdin, we currently assume that `ip=true, iprange=true, cidr=true`, without letting the user change those options, the way we allow it for `inputs`. This causes `range2ips` to output IPs even for CIDR inputs incorrectly - it should rather ignore CIDRs. So, you should modify `lib.IPListWriteFrom` so that we can modify these (currently implicit) booleans for all the different input sources, whether that source is the CLI argument array (i.e. the variable `inputs`), stdin, or some file.